interview-questions.md

```diff
@@ -8,36 +8,117 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: I would generate a new migration file using ruby to add a column to the existing table. An example of this would be add_to_column that can let you add to a certain table so it adds a column to it. The foreign key would belong to the that model so in this case the foreign key would be students and it would belong to Cohort.
+                 
+       
 
-Researched answer:
+Researched answer: An example of this for creating the association for Cohort and students:
 
-2. Which RESTful routes must always be passed params? Why?
+                class Cohort < ApplicationRecord
+                 has_many :students
+                 end
+
+                class Student < ApplicationRecord
+                belongs_to :Cohort
+                end
+
+The foreign key would belong to students as the primary key is Cohort.
 
-Your answer:
+2. Which RESTful routes must always be passed params? Why?
 
-Researched answer:
+Your answer: I'm drawing a blank on the restful routes but the reason the params must be passed is the requirements that are established for that particular table. Params offer control over what data is stored in the database. The attributes that are listed in the params will be added.
+    private 
+    def blog_params
+    params.require(:blog).permit(:title, :content)
+    end
+In this example, the params require the title and content to allow the user to create new data.
+
+Researched answer: The restful routes that must always pass params are show, create, destroy and edit. Parameters allow users to create new information by passing a value into the route. This allows the user to have a user friendly experience rather than having to type directly in to the URL. Adding the routes with params allows the page to be more dynamic. A value is needed to be passed to the param. 
```

Comment:
Exactly, the reason these restful routes require a parameter is to know which data we are targeting.